crates/node/engine/src/task_queue/core.rs

```diff
@@ -31,23 +28,21 @@ pub struct Engine {
     /// A sender that can be used to notify the engine actor of state changes.
     state_sender: Sender<EngineState>,
     /// The task queue.
-    tasks: HashMap<EngineTaskType, VecDeque<EngineTask>>,
-    /// The current task being executed.
-    cursor: EngineTaskType,
+    tasks: VecDeque<EngineTask>,
 }
 
 impl Engine {
     /// Creates a new [`Engine`] with an empty task queue and the passed initial [`EngineState`].
     ///
     /// An initial [`EngineTask::ForkchoiceUpdate`] is added to the task queue to synchronize the
     /// engine with the forkchoice state of the [`EngineState`].
-    pub fn new(initial_state: EngineState, state_sender: Sender<EngineState>) -> Self {
-        Self {
-            state: initial_state,
-            tasks: HashMap::new(),
-            cursor: EngineTaskType::ForkchoiceUpdate,
-            state_sender,
-        }
+    pub const fn new(initial_state: EngineState, state_sender: Sender<EngineState>) -> Self {
+        Self { state: initial_state, tasks: VecDeque::new(), state_sender }
+    }
+
+    /// Returns true if the inner [`EngineState`] is uninitialized.
+    pub fn is_state_uninitialized(&self) -> bool {
+        self.state == EngineState::default()
     }
```

Comment:
Logically, I think I'd prefer the truthy case here and to simple invert it to check if uninitialized.

i.e.
```rust
    /// Returns true if the inner [`EngineState`] is initialized.
    pub fn is_state_initialized(&self) -> bool {
        self.state != EngineState::default()
    }
```